### PR TITLE
did: simplify public API, add missing required algorithms

### PR DESCRIPTION
**Pull request by MichaelMure**
The test vectors from the did:key specification identified incorrect handling of some keys as well as edge cases where the library returns an error instead of returning the DID the user expects.  See `/did/README.md` for more information on executing the test vectors.

### CODE REVIEW

File: did/crypto.go [hunk 1]
```diff
@@ -1,43 +1,173 @@
...
+
+// GenerateEd25519 generates a Secp256k1 private key and the matching DID.
+func GenerateRSA() (crypto.PrivKey, DID, error) {
+	priv, pub, err := crypto.GenerateSecp256k1Key(rand.Reader)
+	if err != nil {
+		return nil, Undef, nil
+	}
+	did, err := FromPubKey(pub)
+	return priv, did, err
+}
```
Comment (fabiobozzo) [inline did/crypto.go:53]: Wait, aren't these two functions "swapped" ?

Comment (smoyer64) [reply did/crypto.go:53]: They are ... should we have those functions at all or just let the user BYOKP (bring your own keypair)
